FormalConjectures/Paper/CasasAlvero.lean

```diff
@@ -0,0 +1,95 @@
+of characteristic zero shares a root with each of its derivatives up to order d-1, then P must
+be of the form (X - α)^d for some α.
+
+The conjecture has been proven for:
+* Degrees d ≤ 8
+* Degrees of the form p^k where p is prime
+* Degrees of the form 2p^k where p is prime
+
+The conjecture is false in positive characteristic p for polynomials of degree d > p.
+
+The conjecture is now claimed to be proven in this paper:
+* [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/2505.18000)
+
+-/
+
+open Polynomial
+
+/--
+A polynomial P satisfies the Casas-Alvero property if it shares a root with each
+of its derivatives up to order d-1, where d is the degree of P.
+-/
+def HasCasasAlveroProp {F : Type} [Field F] (P : F[X]) : Prop :=
+  ∀ i ∈ Finset.range P.natDegree, ∃ α : F, IsRoot P α ∧ IsRoot (derivative^[i] P) α
+
+/--
+The Casas-Alvero conjecture states that in characteristic zero, if a monic polynomial P
+has the Casas-Alvero property, then P = (X - α)^d for some α.
+-/
+@[category research open, AMS 12]
+theorem casas_alvero_conjecture {K : Type} [Field K] [CharZero K] (P : K[X]) (hP : Monic P) :
+    HasCasasAlveroProp P → ∃ α : K, P = (X - C α) ^ P.natDegree := by
+  sorry
+
+/--
+The Casas-Alvero conjecture holds for polynomials of prime power degree.
+This was proved by Graf von Bothmer, Labs, Schicho, and van de Woestijne.
+
+Reference: [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/math/0605090)
+-/
+@[category research solved, AMS 12]
+theorem casas_alvero.prime_power {K : Type} [Field K] [CharZero K]
+    (p k : ℕ) (hp : p.Prime) (P : K[X]) (hP : Monic P) (hd : P.natDegree = p^k) :
+    HasCasasAlveroProp P → ∃ α : K, P = (X - C α) ^ P.natDegree := by
+  sorry
+
+/--
+The Casas-Alvero conjecture holds for polynomials of degree 2p^k where p is prime.
+This was proved by Graf von Bothmer, Labs, Schicho, and van de Woestijne.
+
+Reference: [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/math/0605090)
+-/
+@[category research solved, AMS 12]
+theorem casas_alvero.double_prime_power {K : Type} [Field K] [CharZero K]
+    (p k : ℕ) (hp : p.Prime) (P : K[X]) (hP : Monic P) (hd : P.natDegree = 2 * p^k) :
+    HasCasasAlveroProp P → ∃ α : K, P = (X - C α) ^ P.natDegree := by
+  sorry
+
+/--
+The Casas-Alvero conjecture fails in positive characteristic p for polynomials of degree d > p.
+This was shown by Graf von Bothmer, Labs, Schicho, and van de Woestijne.
+
+Reference: [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/math/0605090)
+-/
+@[category research solved, AMS 12]
+theorem casas_alvero.positive_char_counterexample {p : ℕ} (hp : p.Prime) :
+    ∃ K : Type, ∃ (_ : Field K) (_ : CharP K p) (d : ℕ) (hd : d > p) (P : K[X]),
+      Monic P ∧ P.natDegree = d ∧ HasCasasAlveroProp P ∧
+      ¬∃ α : K, P = (X - C α) ^ P.natDegree := by
```

Comment:
If you want this to work in positive characteristics, you have to use [Polynomial.hasseDeriv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/HasseDeriv.html#Polynomial.hasseDeriv) in the definition of HasCasasAlveroProp.

This statement also has a redundant variable `d`; it could be stated simply as
```suggestion
theorem casas_alvero.positive_char_counterexample {p : ℕ} (hp : p.Prime) :
    ∃ K : Type, ∃ (_ : Field K) (_ : CharP K p) (hd : P.natDegree > p) (P : K[X]),
      Monic P ∧ HasCasasAlveroProp P ∧ ¬∃ α : K, P = (X - C α) ^ P.natDegree := by
```